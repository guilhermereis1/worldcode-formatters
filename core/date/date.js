"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.date = exports.date_time = void 0;
var moment = require("moment");
var date_time = function (dateTimeString) {
    var parsedDate = moment(dateTimeString);
    if (!parsedDate.isValid())
        return "Invalid Date Time";
    return parsedDate.format("DD/MM/YYYY HH:mm:ss");
};
exports.date_time = date_time;
var date = function (dateString) {
    var parsedDate = moment(dateString);
    if (!parsedDate.isValid())
        return "Invalid Date";
    return parsedDate.format("DD/MM/YYYY");
};
exports.date = date;
