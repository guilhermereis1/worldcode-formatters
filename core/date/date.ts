const moment = require("moment");

export const date_time = (dateTimeString: string): string => {
  const parsedDate = moment(dateTimeString);

  if (!parsedDate.isValid()) return "Invalid Date Time";

  return parsedDate.format("DD/MM/YYYY HH:mm:ss");
};

export const date = (dateString: string): string => {
  const parsedDate = moment(dateString);

  if (!parsedDate.isValid()) return "Invalid Date";

  return parsedDate.format("DD/MM/YYYY");
};
