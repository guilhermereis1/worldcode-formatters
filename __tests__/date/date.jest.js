import { date_time, date } from "../../core/date/date"; // Importe o mÃ³dulo que deseja testar

describe("Date and Date Time", () => {
  it("Deve retornar a Data formatada", () => {
    expect(date("2023-09-15 21:19:20.909-04")).toBe("15/09/2023");
  });

  it("Deve retornar a Data e Hora formatada", () => {
    expect(date_time("2023-09-15 21:19:20.909-04")).toBe("15/09/2023 21:19:20");
  });

  it("Deve retornar uma String 'Invalid Date Time'", () => {
    expect(date_time("")).toBe("Invalid Date Time");
  });

  it("Deve retornar a Data formatada", () => {
    expect(date("2023-09-15 21:19:20.909-04")).toMatchSnapshot();
  });

  it("Deve retornar a Data e Hora formatada", () => {
    expect(date_time("2023-09-15 21:19:20.909-04")).toMatchSnapshot();
  });

  it("Deve retornar uma String 'Invalid Date Time'", () => {
    expect(date_time("")).toMatchSnapshot();
  });
});
