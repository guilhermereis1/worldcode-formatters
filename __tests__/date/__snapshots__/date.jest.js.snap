// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date and Date Time Deve retornar a Data e Hora formatada 1`] = `"15/09/2023 21:19:20"`;

exports[`Date and Date Time Deve retornar a Data formatada 1`] = `"15/09/2023"`;

exports[`Date and Date Time Deve retornar uma String 'Invalid Date Time' 1`] = `"Invalid Date Time"`;
